partitura/musicanalysis/performance_codec.py

```diff
@@ -747,21 +747,17 @@ def get_time_maps_from_alignment(
 
     # Remove grace notes
     if remove_ornaments:
-        # TODO: check that all onsets have a duration?
+        # check that all onsets have a duration
         # ornaments (grace notes) do not have a duration
-        score_unique_onset_idxs = np.array(
-            [
-                np.where(np.logical_and(score_onsets == u, score_durations > 0))[0]
-                for u in score_unique_onsets
-            ],
-            dtype=object,
-        )
+        score_unique_onset_idxs = [
+            np.where(np.logical_and(score_onsets == u, score_durations > 0))[0]
+            for u in score_unique_onsets
+        ]
 
     else:
-        score_unique_onset_idxs = np.array(
-            [np.where(score_onsets == u)[0] for u in score_unique_onsets],
-            dtype=object,
-        )
+        score_unique_onset_idxs = [
+            np.where(score_onsets == u)[0] for u in score_unique_onsets
+        ]
```

Comment:
The dtype was set to object before because the output was an array of arrays of different sizes (and numpy used to complain about this).

Regarding your second point, I don't see how that could happen, since it would have a valid onset value.

```python
import numpy as np

note_array = np.array(
    [
        (60, 0.0, 1.0, "n0"),
        (62, 1.0, 1.0, "n1"),
        (63, 2.0, 0.0, "n2"), # this note has duration 0 and is the only one in onset 2
        (64, 2.5, 1.0, "n3"),
    ],
    dtype=[
        ("pitch", int),
        ("onset_beat", float),
        ("duration_beat", float),
        ("id", "U10"),
    ],
)


unique_onsets = np.unique(note_array["onset_beat"])

unique_onset_idxs = [np.where(note_array["onset_beat"]== u)[0] for u in unique_onsets]

# Expected result
# [array([0]), array([1]), array([2]), array([3])]
```


Or am I missing something?